### ISSUE TITLE
Placeholders not showing up in structure mode sidebar

### ISSUE THREAD

**Issue by ghost** [django-cms/django-cms#6570]
### Summary

Static placeholders and placeholders attached to a model via a `PlaceholderField` and rendered using `render_placeholder` stopped showing up in the structure mode sidebar after upgrading to v3.5. Placeholders on regular cms pages still work as expected.

This issue only manifests on url endpoints which have a page node attached, **but** are served by a view other than `cms.views.details`. After some investigation it turns out that this is because the toolbar wrongly selects the `ContentRenderer` instead of the `LegacyRenderer` in such cases. It does so on the assumption (in `BaseToolbar.uses_legacy_structure_mode`) that a truthy value of the `request.current_page` attribute signals that the current request is being serviced by the built-in `cms.views.details` view, which supports the non-legacy structure mode.

We managed to work around the issue by replacing the `CurrentPageMiddleware` with a middleware that sets `request.current_page = None` instead as some of the built-in cms toolbars rely on the attribute being present when they're being populated.

### Expected behaviour

Placeholders available for editing are shown in the sidebar.

### Actual behaviour

The structure mode sidebar is empty even though there are placeholders in the template being rendered.

### Environment

* Python version: 3.5.6
* Django version: 1.11.16
* django CMS version: 3.5.3

**Comment by tariqmostahid** [comment]
Is this issue fixed in 3.6.0?

**Comment by tariqmostahid** [comment]
As discussed in this [thread](https://github.com/divio/django-cms/issues/6527)I fixed this issue by setting my `request.current_page=None` for that view.

**Comment by db654** [comment]
I have a similar/likely related issue on urls that have both a cms page and an application view mounted (not via apphook). This used to work fine but now seems to result in static_placeholders located in the application template not rendering in the structure sidebar. If I remove the cms page, they reappear.

**Comment by DGathagu** [comment]
> Is this issue fixed in 3.6.0?

Nope! sadly they haven't fixed it though its a long overdue fix. Still using version 3.4.6 meself.

**Comment by benzkji** [comment]
I've used normal placeholders to elegantly fill some of my aplications texts, for example confirmation pages that can have images, galleries, whatever, in their own placeholder. All these placeholders seem gone in 3.5+ :|. I know they are still there, but not showing ;-) BUT: I always considered this some kind of hack, and I am not that surprised it suddenly stopped working?!

To be clear: Placeholders are show when on the "root" url of my apphook, but not on any sub views, like /confirmed/, or anything else.

**Comment by creimers** [comment]
I can report that this still happening with `django-cms==3.7.1` and `django==1.11.29`. No `render_placeholder` placeholders in the sidebar on app hook sub pages.

**Comment by GabrielDumbrava** [comment]
Not working for me using the following versions:
```
Django==2.2.11
django-cms==3.7.1
Python 3.7.3
```
I did manage to work around this by setting the CMS_PLACEHOLDER_CONF and using a default text plugin with some dummy text. This allowed me to edit the plugin by double click-ing on it.

What is strange is that other static placeholders, from the home page do show up in the structure toolbar ...

**Comment by thisisrevere** [comment]
Same issue here. Any word on a workaround/fix?

**Comment by GabrielDumbrava** [comment]
I have to add, I don't know if this makes any difference is that the page where my static placeholders don't show up are provided by a custom module that is integrated withing django-cms with an apphook.

**Comment by thisisrevere** [comment]
> I have to add, I don't know if this makes any difference is that the page where my static placeholders don't show up are provided by a custom module that is integrated withing django-cms with an apphook.

Standard Django CMS default page here!

**Comment by foarsitter** [comment]
Adding PlaceholderAdminMixin to the ModelAdmin an re-save the dynamic object with the PlaceholderField solves the problem for me. After reading better it is all there in the docs: http://docs.django-cms.org/en/latest/how_to/placeholders.html?highlight=PlaceholderAdminMixin

**Comment by SteinRobert** [comment]
**TLDR**
Static placeholders within the template used by the `show_menu` template tag do not appear within the structure mode sidebar.

**Explanation**
I also noticed that some StaticPlaceholders, not all, do not appear within the structure mode sidebar.
After digging through some code I found the cause: The static placeholder which wasn't rendered in the structure mode sidebar was part of the template used by the `ShowMenu` template tag.

The function [`_scan_placeholders`](https://github.com/django-cms/django-cms/blob/develop/cms/utils/placeholder.py#L169) which is called by [`_scan_static_placeholders`](https://github.com/django-cms/django-cms/blob/develop/cms/utils/placeholder.py#L234) to find all `StaticPlaceholderNode` instances. The code within `_scan_placeholders` simply cannot cope with the `ShowMenu` node. 
I was thinking, that the `ShowMenu` node or `show_menu` tag should be analyzed in the same way as `include` nodes.
However, the `ShowMenu` node itself has a property called `child_nodelists` which is empty. Therefore `_scan_placeholders` assumes there is nothing to be found.

Not sure if this covers all the other cases described here - I can imagine that some other template tags may cause the same behavior.

**Temporary solution**
 Since, in my case, there are very few plugins to be rendered within the faulty static placeholder I added it within a hidden `div` to my template. This is it is shown within the structure mode sidebar again.

```html
<div class="hidden">
    {% static_placeholder 'my_placeholder' %}
</div>

```

**Comment by i-salameh95** [comment]
the issue appear with me today :( 
totally headache, I'm trying from hours, because I was assume its from my code, but it seems like a bug ! 
I'm using django cms 3.11

**Comment by adlh** [comment]
I don't know if it's caused by the same issue, but the static placeholders are also not being shown in a brand-new django-cms 4.1.0 project.

**Comment by fsbraun** [comment]
@adlh Static placeholders are replaced by aliases in django CMS 4. They have their own editing endpoint from django CMS 4 on.
